### PR TITLE
implement network task for OpenStack platform

### PR DESCRIPTION
**Pull request by zengchen1024**
Implement network task for Openstack platform.
This is part of work of issue (#3566)

### CODE REVIEW

**Comment by chrislovecnm** [comment]
/ok-to-test

We are close to 1.9 release, so we will merge this into 1.10.  Will review the PR soon.

File: upup/pkg/fi/cloudup/openstacktasks/network.go [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+
+func (n *Network) CompareWithID() *string {
+	return n.ID
+}
+
+func (n *Network) Find(context *fi.Context) (*Network, error) {
+	cloud := context.Cloud.(openstack.OpenstackCloud)
+	opt := networks.ListOpts{
+		Name: fi.StringValue(n.Name),
+		ID:   fi.StringValue(n.Name),
```
Comment (justinsb) [inline upup/pkg/fi/cloudup/openstacktasks/network.go]: Should this one be `n.ID`?

File: upup/pkg/fi/cloudup/openstacktasks/network.go [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+		v, err := t.Cloud.CreateNetwork(opt)
+		if err != nil {
+			return fmt.Errorf("Error creating network: %v", err)
+		}
+
+		e.ID = fi.String(v.ID)
+		return nil
+	}
+
+	glog.V(2).Infof("RenderOpenstak::did nothing")
```
Comment (justinsb) [inline upup/pkg/fi/cloudup/openstacktasks/network.go]: FYI typo in OpenStack :-)

**Comment by justinsb** [comment]
/lgtm

Some observations, but LGTM

File: upup/pkg/fi/cloudup/openstacktasks/network.go (hunk 1 above)
Comment (zengchen1024) [reply upup/pkg/fi/cloudup/openstacktasks/network.go]: Thanks for the reviewing. Done

File: upup/pkg/fi/cloudup/openstacktasks/network.go (hunk 2 above)
Comment (zengchen1024) [reply upup/pkg/fi/cloudup/openstacktasks/network.go]: Done

**Pull request merged**